classes/07class/kms/README.md

```diff
@@ -0,0 +1,106 @@
+# KMS
+This class is an introduction to AWS KMS.
+The main goal is to explain the purpose, the benefits and how to use it.
+
+***Contents***
+- [What is AWS KMS?](#what-is-aws-kms)
+- [Purpose of KMS](#purpose-of-kms)
+- [Types of Keys](#types-of-keys)
+- [How to use KMS](#purpose-of-kms)
+- [Keys' Management](#keys-management)
+- [Main AWS Services Integrated with KMS](#main-aws-services-integrated-with-kms)
+
+## What is AWS KMS?
+AWS KMS(Key Management Service) helps you to centrally manage and store you cryptographic keys and control their usae across serveral AWS services and in applications. It's a managed regional service, which means that the HA across AZ is already handled by AWS and keys can only be used inside the region it exists.
+
+
+## Purpose of KMS
+KMS purpose is to encrypt data. Many services in AWS provides an encryption option which will make the data be encrypted when is handled by an AWS service like S3, DynamoDB, EBS and RDS. [Here](https://aws.amazon.com/kms/features/) you can find a list of all services that integrates with KMS. 
+
+This helps enhance the data security, since only people with permission to read the data `and` with access to the key used to encrypt the data will be able to properly read it.
+
+## Types of Keys
+AWS works with two types of keys. The `Symmetric` have a single encryption key that is used for both encryption and decryption operations. The `Asymmetric` have a public and a private key pair that can be used for encrypt/decrypt or sign/verify operations. Asymmetric keys are not available in all regions yet.
+
+For each type of key a list of types and algorithms are supported and can be found in the [KMS FAQ page](https://aws.amazon.com/kms/faqs/).
+
+## How to use KMS?
+KMS is easy to use and in most cases you can encrypt/decrypt your data without having to interact with the KMS service directly. When creating resources(through console, API or terraform) you can specify if encryption is enabled or disabled. In this situation, you're already making use of KMS indirectly.
+
+For example, when creating an EBS volume in terraform:
+```terraform
+resource "aws_ebs_volume" "example" {
+  availability_zone = var.az
+  size              = var.disk_size
+  encrypted         = true
+
+  tags = {
+    Name = var.name
+  }
+}
+```
+In the example above, the EBS volume being created will use a KMS key to be ecnrypted. Since we're not specyifying the key to be used through the `kms_key_id` argument, the resource will be created using the default KMS key for it's service(ebs in this case).
```

Comment:
```suggestion
In the example above, the EBS volume being created will use a KMS key to be encrypted. Since we're not specifying the key to be used through the `kms_key_id` argument, the resource will be created using the default KMS key for its service (EBS in this case).
```